app/main.py

```diff
@@ -1,3 +1,86 @@
-def shop_trip():
-    # write your code here
-    pass
+import datetime
+from app.claasss import Customer, Shops
+from app.fiilee import Config
+
+
+def shop_trip() -> None:
+    config = Config("app/config.json")
+    res_shop = []
+    for elem in config.shops:
+        res_shop.append(Shops(elem["name"],
+                              elem["location"],
+                              elem["products"]))
+
+    res_customer = []
+    for elem in config.customers:
+        res_customer.append(Customer(elem["name"],
+                                     elem["product_cart"],
+                                     elem["location"],
+                                     elem["money"],
+                                     elem["car"]))
+
+    price_for_fuel = config.fuel_price
+    for elem in res_customer:
+
+        print(f"{elem.name} has {elem.money} dollars")
+        res = {}
+        for shop in res_shop:
+            distance = ((elem.location[0] - shop.location[0]) ** 2
+                        + (elem.location[1] - shop.location[1]) ** 2) ** 0.5
+            liters_needed = (distance / 100) * elem.car["fuel_consumption"]
+            price = liters_needed * price_for_fuel
+            money_need = ((elem.product_cart["milk"]
+                           * shop.products["milk"]
+                           + elem.product_cart["bread"]
+                           * shop.products["bread"]
+                           + elem.product_cart["butter"]
+                           * shop.products["butter"])
+                          + price * 2)
+            res[shop.name] = money_need
+            print(f"{elem.name}'s trip to the "
+                  f"{shop.name} costs {round(money_need, 2)}")
+        min_key = min(res, key=res.get)
+        min_value = res[min_key]
+        if elem.money > min_value:
```

Comment:
There is a logical error here. The condition should be `if elem.money >= min_value:` instead of `if elem.money > min_value:` to ensure the customer can afford the exact amount needed for the trip and purchase.